x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/X-VectorBlock.java.st

```diff
@@ -80,15 +83,14 @@ $endif$
 
     @Override
     public boolean isReleased() {
-        return released || vector.isReleased();
+        return super.isReleased() || vector.isReleased();
```

Comment:
I think the ownership here is unclear, but it's perfectly fine. Let's leave it as is.

Reply:
Yeah, I don't like this either.

Currently, the tests do not pass without this, complaining about unclosed blocks. We have a couple of spots where we close vectors directly, but vectors have enclosing blocks (created in the vectors' constructors).

So, IMO; ideally we should refactor at some point so that either
- we always close blocks, not vectors, or
- we stop creating blocks in vectors' constructors - instead, vectors can be completely oblivious to blocks, just being owned by them (my preference).

Reply:
Thanks, Alex! Yes, I think we can also make Vector ref counted and create a new Block in `asBlock` instead of the constructor. But let's do it later.